presto-hive/src/main/java/com/facebook/presto/hive/PrestoS3FileSystem.java

```diff
@@ -444,6 +455,9 @@ protected ObjectListing computeNext(ObjectListing previous)
 
     private Iterator<LocatedFileStatus> statusFromObjects(List<S3ObjectSummary> objects)
     {
+        // NOTE: for encrypted objects, S3ObjectSummary.size() used below is NOT correct,
+        // however, to get the correct size we'd need to make an additional request to get
+        // user metadata, and in this case it doesn't matter.
```

Comment:
Is this "doesn't matter" because the encrypted size is always larger? The Hive connector computes splits based on this size, so if the encrypted size was smaller, it would be possible to not generate a split for part of the file and thus miss some data.